SECP-service/src/main/java/com/visucius/secp/Controllers/GroupController.java

```diff
@@ -0,0 +1,136 @@
+package com.visucius.secp.Controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.visucius.secp.DTO.GroupCreationRequest;
+import com.visucius.secp.daos.GroupDAO;
+import com.visucius.secp.daos.RolesDAO;
+import com.visucius.secp.daos.UserDAO;
+import com.visucius.secp.models.Group;
+import com.visucius.secp.models.Role;
+import com.visucius.secp.models.User;
+import com.visucius.secp.util.InputValidator;
+import com.visucius.secp.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class GroupController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GroupController.class);
+    public static final int MINIMUM_AMOUNT_OF_USERS = 2;
+    public static final int MAXIMUM_AMOUNT_OF_USERS = 50;
+
+    private GroupDAO groupRepository;
+    private UserDAO userRepository;
+    private RolesDAO rolesRepository;
+
+    public GroupController(GroupDAO groupRepository, UserDAO userRepository, RolesDAO rolesRepository)
+    {
+        this.groupRepository = groupRepository;
+        this.userRepository = userRepository;
+        this.rolesRepository = rolesRepository;
+    }
+
+
+    public Response createGroup(GroupCreationRequest request)
+    {
+        List<String> errors = validateCreationRequest(request);
+
+        if(!errors.isEmpty())
+        {
+            try
+            {
+                throw new WebApplicationException(
+                    JsonUtil.convertToJsonString(errors),
+                    Response.Status.BAD_REQUEST);
+            }
+            catch (JsonProcessingException exception)
+            {
+                LOG.error(exception.getMessage());
+                throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
+            }
+        }
+
+        Group group = new Group(request.name);
+        group.setRoles(getRoles(request.roles));
+        group.setUsers(getUsers(request.users));
```

Comment:
this should be a call to the database to figure out users who have the roles and permission matching that of the group.